scripts/checksha.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import subprocess
+import argparse
+from subprocess import CalledProcessError
+
+import ansible.runner
+import ansible.inventory
+
+
+INVENTORY_FILENAME = 'dynamic_inventory.py'
+BASE_PATH = '/opt/stackforge'
+CLONE_DIR = '/tmp'
+
+
+def get_args():
+    '''Get the arguments.'''
+    parser = argparse.ArgumentParser(
+        prog='checksha',
+        description='given a commit sha and a project name, find out if '
+                    'the commit is in the wheel for project currently '
+                    'deployed')
+    parser.add_argument(
+        'project', type=str, default=None,
+        help='name of openstack project wheel to query')
+    parser.add_argument(
+        'commit', type=str, default=None,
+        help='SHA of commit that you are interested in')
+    parser.add_argument(
+        '-s', '--sha', type=str, default=None,
+        help='provide a parent sha rather than interrogating the wheels '
+             'that are deployed in the solution to get the sha')
+
+    return parser.parse_args()
+
+
+def find_inventory(inv, path):
+    '''Find the dynamic inventory file'''
+    for root, dirs, files in os.walk(path):
+        if inv in files:
+            return os.path.join(root, inv)
+
+
+def clone_git_repo(project, path=CLONE_DIR):
+    '''clone a git repo'''
+    os.chdir(path)
+    if not os.path.exists(project):
+        print 'cloning %s project to interrogate the history...' \
+              'please be patient this may take a while' % project
+        try:
+            subprocess.check_call(
+                'git clone https://github.com/openstack/%s.git '
+                '--quiet' % project,
+                shell=True, stderr=subprocess.STDOUT)
+        except CalledProcessError as e:
+            print 'Something went wrong when trying to clone the repo'
+            sys.exit(e)
+
+
+def parse_results(results, project):
+    '''parse and print results of the ansible run'''
+    shas = set()
+    hostnames = []
+    for (hostname, result) in results['contacted'].items():
+        shas.add(result['stdout'].split()[3])
+        hostnames.append(hostname)
+    # bail if multiple or no shas found
```

Comment:
Why exit if multiple shas? A heterogeneous deployment isn't necessary a bad thing. Why not just print a warning and carry on?